terraso_backend/apps/story_map/views.py

```diff
@@ -52,13 +52,28 @@ def post(self, request, **kwargs):
         )
 
         entry_form = StoryMapForm(data=form_data)
-
         if not entry_form.is_valid():
             error_messages = get_error_messages(entry_form.errors.as_data())
             return JsonResponse(
                 {"errors": [{"message": [asdict(e) for e in error_messages]}]}, status=400
             )
 
+        if "chapters" in config:
```

Comment:
Issue: Can you create a separate function to hold this code? I also think you need to run this for `StoryMapUpdateView`